purchase_sale_inter_company/models/sale_order.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2013-Today Odoo SA
+# Copyright 2016 Chafique DELLI @ Akretion
+# Copyright 2018 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    auto_purchase_order_id = fields.Many2one(
+        comodel_name='purchase.order',
+        string='Source Purchase Order',
+        readonly=True,
+        copy=False,
+    )
+
+    @api.multi
+    def action_confirm(self):
+        for order in self.filtered('auto_purchase_order_id'):
+            for line in order.order_line.sudo():
+                if line.auto_purchase_line_id:
+                    line.auto_purchase_line_id.price_unit = line.price_unit
```

Comment:
Wouldn't it be interesting to update the line qty if it's changed in the SO? (and maybe log the change into the PO chatter)